test/common/binders/ResidentPropertyBindersSpec.scala

```diff
@@ -88,4 +89,111 @@ class ResidentPropertyBindersSpec extends UnitSpec with MockitoSugar {
+          "acquisitionValue" -> Seq("2000.0"),
+          "acquisitionCosts" -> Seq("2000.0"),
+          "improvements" -> Seq("2000.0"),
+          "prrValue" -> Seq("2000.0"),
+          "lettingReliefs" -> Seq("2000.0"),
+          "allowableLosses" -> Seq("2000.0"),
+          "broughtForwardLosses" -> Seq("2000.0"),
+          "annualExemptAmount" -> Seq("2000.0"),
+          "disposalDate" -> Seq("2016-10-10")))
+
+        val date = DateTime.parse("2016-10-10")
+        result shouldBe Some(Right(PropertyChargeableGainModel(propertyTotalGainModel, Some(2000.0), Some(2000.0), Some(2000.0),
+                                  Some(2000.0), 2000, date)))
+      }
+
+      "return a valid PropertyTotalGainModel from a valid map with different values" in {
+        val propertyTotalGainModel = new PropertyTotalGainModel(TotalGainModel(2000, 2500, 3000, 3500), 1500)
+        val result = binder.bind("", Map("disposalValue" -> Seq("2000.0"),
+          "disposalCosts" -> Seq("2500.0"),
+          "acquisitionValue" -> Seq("3000.0"),
+          "acquisitionCosts" -> Seq("3500.0"),
+          "improvements" -> Seq("1500.0"),
+          "prrValue" -> Seq("2050.0"),
+          "lettingReliefs" -> Seq("2100.0"),
+          "allowableLosses" -> Seq("2200.0"),
+          "broughtForwardLosses" -> Seq("2300.0"),
+          "annualExemptAmount" -> Seq("2400.0"),
+          "disposalDate" -> Seq("2016-10-10")))
+
+        val date = DateTime.parse("2016-10-10")
+        result shouldBe Some(Right(PropertyChargeableGainModel(propertyTotalGainModel, Some(2050.0), Some(2100.0), Some(2200.0),
+          Some(2300.0), 2400.0, date)))
+
+      }
+
+      "return an error message when one component fails" in {
+        val result = binder.bind("", Map("disposalValue" -> Seq("1000.0"),
+          "disposalCosts" -> Seq("1000.0"),
+          "acquisitionValue" -> Seq("b"),
+          "acquisitionCosts" -> Seq("1000.0"),
+          "improvements" -> Seq("1000.0"),
+          "prrValue" -> Seq("2000.0"),
+          "lettingReliefs" -> Seq("2000.0"),
+          "allowableLosses" -> Seq("2000.0"),
+          "broughtForwardLosses" -> Seq("2000.0"),
+          "annualExemptAmount" -> Seq("2000.0"),
+          "disposalDate" -> Seq("2016-10-10")))
+
+        result shouldBe Some(Left("""Cannot parse parameter acquisitionValue as Double: For input string: "b""""))
+      }
+
+      "return an error message when a component is missing" in {
+        val result = binder.bind("Any", Map("disposalValue" -> Seq("1000.0"),
+          "disposalCosts" -> Seq("1000.0"),
+          "acquisitionValue" -> Seq("3000.0"),
+          "improvements" -> Seq("1000.0"),
+          "prrValue" -> Seq("2000.0"),
+          "lettingReliefs" -> Seq("2000.0"),
+          "allowableLosses" -> Seq("2000.0"),
+          "broughtForwardLosses" -> Seq("2000.0"),
+          "annualExemptAmount" -> Seq("2000.0"),
+          "disposalDate" -> Seq("2016-10-10")))
```

Comment:
Just commenting so we keep a track.  We could re-factor this to have a single map and then use mac's testing style to re-write this.